### DISCUSSION TITLE
How can I configure my next.js application to use Google login through the same OAuth provider configured in Directus?

### DISCUSSION THREAD

**Discussion by 2820402** [directus/directus#23790] [Feature Requests]
### Summary

How can I configure my next.js application to use Google login through the same OAuth provider configured in Directus? Specifically, what steps should I follow to integrate the authentication flow and ensure my next.js application utilizes the existing OAuth configuration in Directus?

I appreciate any guidance, code examples, or relevant documentation you can share. Thank you in advance!

### Motivation

.

### Detailed Design

.

### Must Have:
- A
- B

### Should Have:
- C

### Could Have:
- D

### Won't Have:
- E

### Drawbacks

.

### Alternatives

.

### Adoption Strategy

.

### Unresolved Questions

.

**Comment by pemontto** [comment]
https://docs.directus.io/self-hosted/sso.html#seamless-sso

From kian on discord

**Step 1**
Add a simple link to your Frontend page
```html
<a href={`${process.env.NEXT_PUBLIC_BACKEND_URL}/auth/login/google?redirect=${process.env.NEXT_PUBLIC_APP_URL}`}>
```

**Step 2**
Setup the SDK. You also need to build your directusSchema
```ts
import { authentication, createDirectus, rest } from '@directus/sdk';
import { DirectusSchema } from '@/lib/directus/schema';

const directus = createDirectus<DirectusSchema>(process.env.NEXT_PUBLIC_BACKEND_URL || '')
  .with(authentication('session', { credentials: 'include' }))
  .with(
    rest({
      credentials: 'include',
      onRequest: (options) => ({ ...options, cache: 'no-store' }),
    })
  );

export default directus;
```

**Reply by ymys** [reply]
That's it? 

I try implementing this, but after choose one google account, it always redirect back to login page. Last time I checked, there were not values at the cookies for access_token etc. Am I missing something here? Please help.
